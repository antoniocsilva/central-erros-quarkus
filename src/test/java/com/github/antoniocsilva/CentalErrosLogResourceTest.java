package com.github.antoniocsilva;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CentralErrosTestLifecycleManager.class)
public class CentalErrosLogResourceTest {

    @Test
    @DataSet("datasets/central-erros-cenario-1.yml")
    public void testBuscarErros(){
        String resultado = given()
                .when().get("/errors")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }

}
