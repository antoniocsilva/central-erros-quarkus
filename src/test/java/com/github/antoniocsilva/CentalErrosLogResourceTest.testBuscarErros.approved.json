{
  "ERROR": 1,
  "INFO": 1
}